tests/common/docker_image.py

```diff
@@ -337,6 +337,53 @@ def wait(self, max_attempts: int = 15, sleep_rate: float = 1.0) -> bool:
         return False
 
 
+class SOEFDockerImage(DockerImage):
+    """Wrapper to SOEF Docker image."""
+
+    def __init__(
+        self, client: DockerClient, addr: str, port: int = 9002,
+    ):
+        """
+        Initialize the SOEF Docker image.
+
+        :param client: the Docker client.
+        :param addr: the address.
+        :param port: the port.
+        """
+        super().__init__(client)
+        self._addr = addr
+        self._port = port
+
+    @property
+    def tag(self) -> str:
+        """Get the image tag."""
+        return "fetchai/soef:latest"
+
+    def _make_ports(self) -> Dict:
+        """Make ports dictionary for Docker."""
+        return {f"{self._port}/tcp": ("0.0.0.0", self._port)}  # nosec
+
+    def create(self) -> Container:
+        """Create the container."""
+        container = self._client.containers.run(
+            self.tag, detach=True, ports=self._make_ports()
+        )
+        return container
+
+    def wait(self, max_attempts: int = 15, sleep_rate: float = 1.0) -> bool:
+        """Wait until the image is up."""
+        request = dict(jsonrpc=2.0, method="web3_clientVersion", params=[], id=1)
+        for i in range(max_attempts):
+            try:
+                response = requests.post(f"{self._addr}:{self._port}", json=request)
+                enforce(response.status_code == 200, "")
+                return True
+            except Exception:
+                logger.info(f"Attempt {i} failed. Retrying in {sleep_rate} seconds...")
+                time.sleep(sleep_rate)
+        return False
+
```

Comment:
here we should devise a health check mechanism for SOEF; that is, we should detect when the SOEF is ready to work.

Reply:
Good point. Should be simple to do. SOEF has an endpoint for this, just need to find out the details.